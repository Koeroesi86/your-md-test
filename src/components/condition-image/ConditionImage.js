import React, { useState, Fragment } from 'react'
import './ConditionImage.scss'

function ConditionImage({ src, label }) {
  const [failedLoad, setFailedLoad] = useState(true);
  return (
    <Fragment>
      {typeof src === 'string' && (
        <img
          className={`image ${failedLoad ? 'hide' : 'show'}`}
          src={src}
          alt={label || ''}
          onError={() => setFailedLoad(true)}
          onLoad={() => setFailedLoad(false)}
        />
      )}
      {failedLoad && (
        <svg
          className={`image show`}
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
          viewBox="0 0 228 220"
        >
          <defs>
            <path id="a" d="M0 0h228v220H0V0z"/>
            <clipPath id="b">
              <use xlinkHref="#a"/>
            </clipPath>
          </defs>
          <use fill="#D8D8D8" xlinkHref="#a"/>
          <g clipPath="url(#b)">
            <use fill="none" stroke="#979797" strokeWidth="2" xlinkHref="#a"/>
          </g>
        </svg>
      )}
    </Fragment>
  );
}

export default ConditionImage
