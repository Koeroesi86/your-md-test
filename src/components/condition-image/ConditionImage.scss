.ConditionImage {
  display: block;

  .image {
    width: 100%;
    height: auto;

    &.show {
      display: block;
    }

    &.hide {
      display: none;
    }
  }
}
