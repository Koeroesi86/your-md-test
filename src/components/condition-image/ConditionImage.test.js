import React from 'react'
import { shallow } from 'enzyme'
import ConditionImage from './ConditionImage'

describe('The GithubLink component', () => {
  it('renders', () => {
    const result = shallow(<ConditionImage/>);

    expect(result).toMatchSnapshot();
  });
});
