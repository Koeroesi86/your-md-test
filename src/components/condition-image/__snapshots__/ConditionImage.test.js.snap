// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GithubLink component renders 1`] = `
<Fragment>
  <svg
    className="image show"
    viewBox="0 0 228 220"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path
        d="M0 0h228v220H0V0z"
        id="a"
      />
      <clipPath
        id="b"
      >
        <use
          xlinkHref="#a"
        />
      </clipPath>
    </defs>
    <use
      fill="#D8D8D8"
      xlinkHref="#a"
    />
    <g
      clipPath="url(#b)"
    >
      <use
        fill="none"
        stroke="#979797"
        strokeWidth="2"
        xlinkHref="#a"
      />
    </g>
  </svg>
</Fragment>
`;
